internal/controller/server/services.go

```diff
@@ -23,40 +27,44 @@ func NewProcServiceServer(exec execution.Execution) protobuf.ClientCPServicesSer
 }
 
 func (s *clientCPServicesServer) PostMetadata(ctx context.Context, request *protobuf.RequestToPostMetadata) (*protobuf.MetadataName, error) {
-	uid, err := id_generator.NextID()
+	uid, err := idgen.NextID()
 	if err != nil {
-		logger.Error(err, "Error while assigning is to the request")
+		log.Error(err, "error while assigning is to the request")
 	}
-	ctx = context.WithValue(ctx, "uid", uid)
+	// should not use basic type string as key in context.WithValue
+	ctx = context.WithValue(ctx, uidKey, uid)
 	name, err := s.procExec.SaveMetadataToDb(ctx, request.Metadata)
-	logger.Error(err, fmt.Sprintf("%v Job Create Request Received - Posting Metadata to etcd", uid))
+	log.Error(err, fmt.Sprintf("%v Job Create Request Received - Posting Metadata to etcd", uid))
 	return name, err
 }
 
 func (s *clientCPServicesServer) GetAllMetadata(ctx context.Context, request *protobuf.RequestToGetAllMetadata) (*protobuf.MetadataArray, error) {
 	dataList, err := s.procExec.ReadAllMetadata(ctx)
-	logger.Error(err, "Getting Metadata")
+	log.Error(err, "Getting Metadata")
 	return dataList, err
 }
 
 func (s *clientCPServicesServer) GetStreamLogs(request *protobuf.RequestForStreamLog, stream protobuf.ClientCPServices_GetStreamLogsServer) error {
 
-	uid, err := id_generator.NextID()
+	uid, err := idgen.NextID()
 	if err != nil {
-		logger.Error(err, "Error while assigning is to the request")
+		log.Error(err, "Error while assigning is to the request")
 	}
-	logString := &protobuf.Log{RequestId: uid, Log: "lorem ipsum logger logger logger dumb"}
+	// RequestId is missing.
```

Comment:
RequestId and Log are from .pb.go files, it will be generated when the make build is executed, because we are not pushing .pb.go files to the repo, instead we are just pushing proto files, and our make build has instructions to generate .pb.go files.

Reply:
Any reason why?. I prefer pushing .pb.go files because if the file is missing that will causing a compile-time error. and I see no bad reason to push the .pb.go files.

Reply:
Actually, we were told that build or output files should not be pushed, and we considered pb.go as build files. Anyways, pb.go is sort of dependency for other files, so I guess pushing them will make more sense. Noted.

Reply:
Yes, things that need to ignored actually are binary and test file related output.